src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1184,12 +1184,23 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::CopySelectionToClipboard(bool trimTrailingWhitespace)
     {
         // extract text from buffer
-        const auto copiedData = _terminal->RetrieveSelectedTextFromBuffer(trimTrailingWhitespace);
+        const auto bufferData = _terminal->RetrieveSelectedTextFromBuffer(trimTrailingWhitespace);
 
+        // convert text: vector<string> --> string
+        std::wstring textData;
+        for (const auto& text : bufferData.text)
+        {
+            textData += text;
+        }
+
+        // convert text to HTML format
+        const auto htmlData = TextBuffer::GenHTML(bufferData, _actualFont.GetUnscaledSize().Y, _actualFont.GetFaceName(), "Windows Terminal");
+        
         _terminal->ClearSelection();
 
         // send data up for clipboard
-        _clipboardCopyHandlers(copiedData);
+        auto copyArgs = winrt::make_self<CopyToClipboardEventArgs>(winrt::to_hstring(textData.c_str()), winrt::to_hstring(htmlData));

```

Comment:
`textData` is a `wstring`, does it need to be wrapped in `to_hstring`? I thought there was an implicit conversion between those guys.

Reply:
So this is dumb. Here's a table of what happened:

|  | `<var-name>` | `.c_str()` | `winrt::to_hstring(<var-name>)` | `winrt::to_hstring(<var-name>.c_str())` |
|--|--|--|--|--|
| `wstring textData` | ❌ | ✔ | ❌ | ✔ |
| `string htmlData` | ❌ | ❌ | ✔ | ✔ |

So, for consistency and to make it look normal, I'm just going with the last one and wrapping in `winrt::to_hstring()` AND calling `.c_str()`.

Reply:
From Dustin: use
`{ <var-name>.data(), <var-name>.size() }`

winrt needs the size to construct the hstring. It's annoying that we need to convert to a `c_str()` or `data()` because we're going from knowing the size, to not knowing the size, so that the hstring constructor finds the size, to construct an hstring...yes, you read that right.

Reply:
except for std::string. Explicitly use to_hstring because we're changing number of bits.

Reply:
https://github.com/microsoft/xlang/issues/553